### PR TITLE
[air] Example for Comet ML

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

After #24459, this PR will add similar support for model artifact saving and an example for experiment tracking with Ray AIR for Comet ML.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by Yard1 — APPROVED** [review]
Nice!

File: python/ray/tune/integration/comet.py [hunk 1]
```diff
@@ -199,6 +209,33 @@ def log_trial_result(self, iteration: int, trial: "Trial", result: Dict):
         for k, v in episode_logs.items():
             experiment.log_curve(k, x=range(len(v)), y=v, step=step)
 
+    def log_trial_save(self, trial: "Trial"):
+        if self.save_checkpoints and trial.checkpoint:
+            if trial not in self._trial_experiments:
+                self.log_trial_start(trial)
```
Comment (amogkam) [inline python/ray/tune/integration/comet.py:215]: Is this something that is required? Seems like we're doing this in `log_trial_result` as well.

Out of scope for this PR, but this logic seems like it shouldn't be the responsibility of the callback. Shouldn't the trial scheduler make sure that the trial is started before trying to log to it?

File: python/ray/tune/integration/comet.py [hunk 2]
```diff
@@ -199,6 +209,33 @@ def log_trial_result(self, iteration: int, trial: "Trial", result: Dict):
...
+                rel_root = os.path.relpath(root, checkpoint_root)
+                for file in files:
+                    local_file = os.path.join(checkpoint_root, rel_root, file)
+                    logical_path = os.path.join(rel_root, file)
+
+                    # Strp leading `./`
+                    if logical_path.startswith("./"):
+                        logical_path = logical_path[2:]
+
+                    artifact.add(local_file, logical_path=logical_path)
```
Comment (amogkam) [inline python/ray/tune/integration/comet.py:235]: Oh man, does Comet have no support for logging directories as artifacts?

File: .buildkite/pipeline.ml.yml [hunk 3]
```diff
@@ -5,10 +5,10 @@
     - DATA_PROCESSING_TESTING=1 INSTALL_HOROVOD=1 ./ci/env/install-dependencies.sh
     - bazel test --config=ci $(./ci/run/bazel_export_options) --build_tests_only --test_tag_filters=-gpu,-needs_credentials python/ray/ml/...
     # Only setup credentials in branch builds
-    - if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then exit 0; fi
-    - python ./ci/env/setup_credentials.py wandb
+    # - if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then exit 0; fi
```
[the remark this answers is not in the document]
Comment (krfricke) [reply .buildkite/pipeline.ml.yml:8]: This is for testing on the PR build and will be reverted if all tests pass

File: python/ray/tune/integration/comet.py (hunk 2 above)
Comment (krfricke) [reply python/ray/tune/integration/comet.py:235]: Unfortunately not...

File: ci/env/setup_credentials.py [hunk 4]
```diff
@@ -14,7 +17,16 @@ def get_and_write_wandb_api_key(client):
         fp.write(f"machine api.wandb.ai\n" f"  login user\n" f"  password {api_key}\n")
 
 
-SERVICES = {"wandb": get_and_write_wandb_api_key}
+def get_and_write_comet_ml_api_key(client):
+    api_key = client.get_secret_value(SecretId=AWS_COMET_SECRET_ARN)["SecretString"]
+    with open(os.path.expanduser("~/.comet.config"), "w") as fp:
+        fp.write(f"[comet]\napi_key={api_key}\n")
```
[the remark this answers is not in the document]
Comment (krfricke) [reply ci/env/setup_credentials.py:23]: Can't save to environment as it's a separate script and not propagated between different bazel tasks.

We're setting up credentials here to use in other tests, I'll add a comment for this later

**Review by amogkam — APPROVED** [review]

**Review by xwjiang2010 — APPROVED** [review]

**Review by xwjiang2010 — APPROVED** [review]

**Pull request merged**